### PR TITLE
First yoman generator-langium test #431

### PR DESCRIPTION
**Pull request by dhuebner**
Add tests to make sure generator-langium produces compilable result

### CODE REVIEW

File: packages/generator-langium/package.json [hunk 1]
```diff
@@ -38,7 +38,9 @@
   },
   "devDependencies": {
     "@types/lodash": "^4.14.170",
-    "@types/yeoman-generator": "^5.0.0"
+    "@types/yeoman-generator": "^5.0.0",
+    "@types/yeoman-test": "^4.0.3",
+    "yeoman-test": "^7.0.0"
```
Comment (spoenemann) [inline packages/generator-langium/package.json:43]: Please add a `"test": "vitest run"` script so we can run this directly from the package folder

**Review by spoenemann — APPROVED** [review]
Wow, yeoman-test has quite heavy dependencies 😅 

It seems this also covers the case where compiling the generated package fails:

> Error: Command failed with exit code 1: npm run build

**Comment by spoenemann** [comment]
@dhuebner I found my old comment again: https://github.com/langium/langium/issues/431#issuecomment-1057772654

Does the test already use the workspace version of langium?

**Comment by dhuebner** [comment]
@spoenemann 
> Wow, yeoman-test has quite heavy dependencies 😅

Yes, I also was very surprised. It's also a long running one compared to other tests :( 

> Does the test already use the workspace version of langium?

I can only confirm that the test failed before your commit here: 
https://github.com/langium/langium/commit/d2760bb3c081c4c5d34132ad38095cda44b64e40
But I will look more deeply into what happens internally. A good start would be to change something locally and see how the test does.

**Comment by dhuebner** [comment]
@spoenemann 
It seems like the test runs against the npm registry. Generator just spawns `npm install` in install phase... 
It might be possible to tweak the test using `yeoman-environment` API so it uses local langium packages, but this needs more investigation and time. I will merge the current state as it tests other important parts of the generator already.

**Pull request merged**